app/main.py

```diff
@@ -1,2 +1,98 @@
+from __future__ import annotations
+from typing import Hashable, Any, Generator
+
+
+class KeyValuePair:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self._key = key
+        self.value = value
+
+    def __eq__(self, other: KeyValuePair | int):
+        if isinstance(other, KeyValuePair):
+            return other.key == self.key
+        return other == self.key
+
+    def __hash__(self):
+        return hash(self.key)
+
+    def __str__(self) -> str:
+        return f"Key={self.key} : Value={self.value}"
+
+    @property
+    def key(self):
+        return self._key
+
+
+# I implement dictionary via chaining
 class Dictionary:
-    pass
+    _INITIAL_CAPACITY = 8
+    _LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self._hash_table = [[] for _ in range(Dictionary._INITIAL_CAPACITY)]
+        self._size = 0
+
+    def _ensure_capacity(self) -> None:
+        self._size += 1
+        if self._size > int(self.capacity * Dictionary._LOAD_FACTOR):
+            self._resize()
+
+    def _resize(self) -> None:
+        new_hash_table = [[] for _ in range(self.capacity * 2)]
+        for pair in iter(self):
+            ind = hash(pair) % len(new_hash_table)
```

Comment:
I haven't ever seen `ind` as abbreviation for `index` in Python code. I'd suggest to use the whole word if you aren't completely sure it's necessary. It only saves you 2 out of 5 characters anyway.